views/js/controller/runner/runner.js

```diff
@@ -171,6 +171,24 @@ define([
                 }
             };
 
+            /**
+             * Handles warnings
+             * @param {String} [displayMessage] - an alternate message to display
+             */
+            var onWarning = function onWarning(err, displayMessage) {
```

Comment:
`onError` and `onWarning` are very similar. Could you refactor them in order to share the same mechanism between the rwo?

Reply:
refactored